### PR TITLE
Create groups endpoint

### PR DESCRIPTION
**Pull request by amandanguyeen**
Started off with groups endpoint but delved into events quite a bit
1. Create group
2. Get group by id
3. Create event 
4. Update event's budget
5. Get event by id 

I saw Youssef was able to get group to auto generate an `id` field (not _id which is ObjectId) but I have no idea how to generate it. To get around it, I just hardcoded an event_id that increments when theres a creation

### CODE REVIEW

**Comment by cuongkoho** [comment]
do you want to merge into `dev` instead of `activity`?

File: activities/migrations/0002_auto_20210924_1241.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    operations = [
+        migrations.CreateModel(
+            name='Activity',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=255)),
+                ('cost', models.IntegerField()),
+                ('description', models.TextField()),
+                ('rating', models.IntegerField()),
+            ],
```
Comment (cuongkoho) [inline activities/migrations/0002_auto_20210924_1241.py]: missing Image

File: event/migrations/0001_initial.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Event',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('event_id', models.IntegerField()),
+                ('name', models.CharField(max_length=200)),
+                ('activities', models.CharField(max_length=9000)),
```
Comment (cuongkoho) [inline event/migrations/0001_initial.py:20]: is this a CharField? I thought this is going to be a dictionary of activities with rating, or an array of (activity,rating)?

File: event/migrations/0002_event_budget.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('event', '0001_initial'),
+    ]
+
+    operations = [
+        migrations.AddField(
```
Comment (cuongkoho) [inline event/migrations/0002_event_budget.py:13]: why do we need a Budget model?

File: event/models.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
+from django.db import models
+
+# Create your models here.
+class Event(models.Model):
+    event_id = models.IntegerField()
```
Comment (cuongkoho) [inline event/models.py]: there's a models.primaryKey you could use iirc, it'll auto increase the number for you

File: event/models.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    name = models.CharField(max_length=200)
+    activities = models.CharField(max_length=9000)
+    budget = models.IntegerField(default=1)
+
+
+    def save(self):
```
Comment (cuongkoho) [inline event/models.py:14]: usually when you create a new event, you don't want supply the ID into creation method, so I think this method doesn't need to be here

File: event/views.py [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        except Exception as e:
+            response = json.dumps({'Error': str(e)})
+        return HttpResponse(response, content_type='text/json')
+
+@csrf_exempt
+def create_event(request):
+    if request.method == 'POST':
+        payload = json.loads(request.body)
+        event_name_input = payload["event_name"]
+        group_id_input = payload["group_id"]
```
Comment (cuongkoho) [inline event/views.py:27]: I don't think we want to pass `group_id` as part of the payload

**Comment by amandanguyeen** [comment]
> do you want to merge into `dev` instead of `activity`?

ah yes it was branched off alex's previous PR i'll change it now

**Review by cuongkoho — CHANGES_REQUESTED** [review]

File: activities/migrations/0002_auto_20210924_1241.py (hunk 1 above)
Comment (amandanguyeen) [reply activities/migrations/0002_auto_20210924_1241.py]: Joe is branching to fix this model

File: event/migrations/0001_initial.py (hunk 2 above)
Comment (amandanguyeen) [reply event/migrations/0001_initial.py:20]: it'll be a stringified list of lists, should it be textfield instead?

File: event/migrations/0002_event_budget.py (hunk 3 above)
Comment (amandanguyeen) [reply event/migrations/0002_event_budget.py:13]: i believe budget is just an int

File: event/models.py (hunk 4 above)
Comment (amandanguyeen) [reply event/models.py]: i used that before but wasn't working properly for some reason :(

File: event/models.py (hunk 5 above)
Comment (amandanguyeen) [reply event/models.py:14]: Right so supplying the id here was the work around for the auto int not working

File: event/views.py (hunk 6 above)
Comment (amandanguyeen) [reply event/views.py:27]: we need to be able to tell the event what group they should belong to no?

**Pull request merged**